test/tester.go

```diff
@@ -78,11 +78,14 @@ func NewDatabaseWithDeps(ctx context.Context, dbName string, deps lib.Dependenci
 
 // NewAccountsTester creates and starts a new AccountsTester service.
 // Use the Close method for a graceful shutdown.
-func NewAccountsTester(dbName string, deps lib.Dependencies) (*AccountsTester, error) {
+func NewAccountsTester(dbName string, promoter api.Promoter, deps lib.Dependencies) (*AccountsTester, error) {
 	// Make sure we have valid dependencies.
 	if deps == nil {
 		deps = &lib.ProductionDependencies{}
 	}
+	if promoter == "" {
```

Comment:
then here, the custom handler requires none nil values for all of these and `NewAccountsTester` just calls the custom handler with defaults.